digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LanePartitioner.scala

```diff
@@ -148,9 +161,16 @@ object LanePartitioner {
   //Groups lanes by roadIdentifier, sideCode, LaneCode, and other lanes on link.
   //Makes sure that all the lanes in group are connected and there are no gaps in lane selection
   def groupMainLanes(lanes: Seq[PieceWiseLane], allLanes: Seq[PieceWiseLane], roadLinks: Map[Long, RoadLink]): Seq[Seq[PieceWiseLane]] = {
-    val lanesGrouped = lanes.groupBy(lane => {
-      val roadLink = roadLinks.get(lane.linkId)
-      val roadIdentifier = roadLink.flatMap(_.roadIdentifier)
+    val (mainLanesWithCutAdditionalLanes, mainLanesWithFullLenghtLanes) = lanes.partition(lane => {
+      val roadLink = roadLinks(lane.linkId)
+      val roadLinkLength = Math.round(roadLink.length * 1000).toDouble/1000
```

Comment:
onkos kertaa 1000 tarpeelista jos homma jaetaan 1000 lopulta? Miksi näin tehdään? commentti vois olla paikalla

Reply:
Noissa pituuksissa on x määrä desimaaleja. Math.round(roadLink.length * 1000).toDouble/1000 rajaa arvon neljään desimaaliin. Pitää tehä näin jos haluaa verrata tielinkin pituutta ja kaistan loppu m-arvoa koska viimeset desimaalit harvoin täsmäävät. Puhutaan siis luokkaa 0.0001 tason epätarkkuudesta